state/protocol/protocol_state/state/protocol_state.go

```diff
@@ -103,34 +95,209 @@ func NewMutableProtocolState(
 ) *MutableProtocolState {
```

Comment:
`NewMutableProtocolState` constructor 
* wraps low-level `storage.ProtocolKVStore` into higher-level `protocol_state.ProtocolKVStore` 
* and initializes hard-coded list of state machine factories 

the logic of constructing the `MutableProtocolState` is delegated to the  non-exported constructor below (that is also used for testing)